apps/gdalalg_raster_contour.h

```diff
@@ -0,0 +1,84 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster contour" subcommand
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#ifndef GDALALG_RASTER_CONTOUR_INCLUDED
+#define GDALALG_RASTER_CONTOUR_INCLUDED
+
+#include "gdalalgorithm.h"
+
+//! @cond Doxygen_Suppress
+
+/************************************************************************/
+/*                     GDALRasterContourAlgorithm                       */
+/************************************************************************/
+
+class GDALRasterContourAlgorithm final : public GDALAlgorithm
+{
+  public:
+    static constexpr const char *NAME = "contour";
+    static constexpr const char *DESCRIPTION =
+        "Creates a vector contour from a raster elevation model (DEM).";
+    static constexpr const char *HELP_URL =
```

Comment:
Height/elevation may be the most common use case, but the algorithm does not know if the data contains heights, temperatures, pressures or whatever. "Elevation" appears throughout in the files and I think it can be so, but a user documentation should mention other sort of interpretations of the source data.

Reply:
"gdal raster contour" feels natural to me, but will the other "dem" tools be similarly "gdal raster hillshade" and so on, even they produce raster output?

I did not find where this --usage comes, but the output is vector even here.

> Positional arguments:
>   -i, --input <INPUT>                                  Input raster dataset [required]
>   -o, --output <OUTPUT>                                Output raster dataset (created by algorithm) [required]

Reply:
It's consistent with https://gdal.org/en/stable/programs/gdal_contour.html though. 
We should change that too.